woocommerce/woocommerce-ios
# [Hazmat Shipping] Handle hazmat category selection

Closes #10584

Why
==========
During the Hazmat shipping declaration, it's mandatory to describe in which category the hazardous material falls in. So whenever the Hazmat section is expanded, we should allow the merchant to select the correct category.

This PR introduces the Hazmat category selection UI interaction, but still with no integration with models and network operations. So when we leave the package details view, the selection is lost.

How
==========
Introduces the hazmat category selection flow that's not connected to no models or network operation yet by defining a Button layout containing a flexible area to allow displaying which hazmat category is currently selected, or asking the merchant to select one.

Screen Capture
==========
https://github.com/woocommerce/woocommerce-ios/assets/5920403/a7908a53-ee29-4d76-a1fd-c5d055160f45

How to Test
==========
1. Open the order details of an order eligible for Shipping label creation
2. Start the shipping label creation flow until you reach the Package details section
3. Inside the package details view, make sure the Contains hazardous materials toggle is visible and interactable
4. Make sure the toggle expands and collapses a basic view that will serve as the main control options for the HAZMAT declaration
5. Hit the `Hazardous material category` section, make sure it works as expected for a selection field, and allow selecting or changing the category any time.

Update release notes:
